File: Python/open_weather_map.py

import json
import pyowm

# APIs information.
open_weather_map_key = ''

class open_weather_map:
    def get_city(self):
        """
        Obter a cidade.
        Author: Gugatb
        Date: 16/06/2018
        Return: city a cidade
        """
        return self.city

    def get_key(self):
        """
        Obter a chave.
        Author: Gugatb
        Date: 16/06/2018
        Return: key a chave
        """
        return self.key

    def get_temperature_by_coordinate(self, latitude, longitude):
        """
        Obter a temperatura.
        Author: Gugatb
        Date: 16/06/2018
        Param: latitude a latitude
        Param: longitude a longitude
        Return: temperature a temperatura
        """
        temperature = ''

        if self.key != '':
            observation = self.owm.weather_at_coords(latitude, longitude)
            weather = observation.get_weather()
            temperature = weather.get_temperature('celsius')
        return temperature

    def get_temperature_by_country(self, city, country):
        """
        Obter a temperatura.
        Author: Gugatb
        Date: 16/06/2018
        Param: city a cidade
        Param: country o pais
        Return: temperature a temperatura
        """
        temperature = ''

        if self.key != '':
            try:
                self.city = city
                self.country = country

                observation = self.owm.weather_at_place(self.city + ',' + self.country)
                weather = observation.get_weather()
                temperature = weather.get_temperature('celsius')
            except:
                self.city = ''
                self.country = ''
        return temperature

    def __init__(self):
        try:
            self.key = open_weather_map_key
            self.owm = pyowm.OWM(self.key)
        except:
            self.key = ''

File: Ruby/open_weather_map.rb

require 'open_weather'

# APIs information.
@open_weather_map_key = ''

# Definir as opcoes.
@options = {units: "metric", APPID: @open_weather_map_key}

# Obter a temperatura.
# Author: Gugatb
# Date: 17/06/2018
# Param: latitude a latitude
# Param: longitude a longitude
# Return: temperature a temperatura
def get_temperature_by_coordinate(latitude, longitude)
    temperature = 0

    begin
      # current = OpenWeather::Current.geocode(latitude, longitude, @options)
      # json = JSON.parse(current.to_json)
      # temperature = Float(json["main"]["temp"])
    rescue Exception => message
      puts message
    end
    return temperature
end

# Obter a temperatura.
# Author: Gugatb
# Date: 17/06/2018
# Param: city a cidade
# Param: country o pais
# Return: temperature a temperatura
def get_temperature_by_country(city, country)
    temperature = 0

    begin
      # current = OpenWeather::Current.city(city + ',' + country, @options)
      # json = JSON.parse(current.to_json)
      # temperature = Float(json["main"]["temp"])
    rescue Exception => message
      puts message
    end
    return temperature
end
